Google.Api.Gax.Grpc/Rest/RestChannel.cs

```diff
@@ -83,10 +87,23 @@ private async Task<ReadHttpResponseMessage> SendAsync<TRequest>(RestMethod restM
             }
             httpRequest.Headers.Add(VersionHeaderBuilder.HeaderName, RestVersion);
 
-            // TODO: How do we cancel this?
-            await AddAuthHeadersAsync(httpRequest, restMethod).ConfigureAwait(false);
+            var addAuthHeadersTask = AddAuthHeadersAsync(httpRequest, restMethod);
+            var delayTask = Task.Delay(-1, cancellationToken);
+            if (delayTask == await Task.WhenAny(addAuthHeadersTask, delayTask).ConfigureAwait(false))
+            {
+                throw new InvalidOperationException("Timeout was reached when waiting for auth headers to be added for a method {restMethod.FullName}.");
```

Comment:
Throw OperationCanceledException instead of InvalidOperationException?

Reply:
I changed to the TimeoutException.